### PR TITLE
complete task 3.1

### PR DESCRIPTION
**Pull request by NguyenChanhTiep**

### CODE REVIEW

File: app/src/main/java/com/framgia/movie/data/model/Movie.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package com.framgia.movie.data.model;
+
+import java.util.List;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/model/Movie.java]: ERROR: Unused import: java.util.List. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/movie/data/remote/MovieApi.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.framgia.movie.data.remote;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class MovieApi {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/MovieApi.java]: WARNING: Utility classes should not have a public or default constructor.

File: app/src/main/java/com/framgia/movie/data/remote/ParceJson.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by TiepNguyen on 25-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class ParceJson {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/ParceJson.java]: WARNING: Utility classes should not have a public or default constructor.

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    @Override
+    public int getItemCount() {
+        return mMovieList.size();
+    }
+
+    static class ViewHolderItem extends RecyclerView.ViewHolder {
+        private ImageView mImageMovie;
+        private TextView mTextMovie;
+
+        public ViewHolderItem(View itemView) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                    .placeholder(R.drawable.image_placeholder_movie)
+                    .into(mImageMovie);
+            mTextMovie.setText(movie.getTitle());
+        }
+    }
+
+    static class ViewHolderLoading extends RecyclerView.ViewHolder {
+        private ProgressBar mProgressBar;
+
+        public ViewHolderLoading(View itemView) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import com.framgia.movie.data.model.Movie;
+import java.util.List;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public interface MainIFace {
+    void DownloadSuccess(String data);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java]: ERROR: Method names should start with a lower case letter (e.g. getWidth())

File: app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    void ParcelJsonSuccess(List<Movie> movieList);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java]: ERROR: Method names should start with a lower case letter (e.g. getWidth())

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @Override
+    protected String doInBackground(String... strings) {
+        try {
+            URL url = new URL(strings[0]);
+            HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
+            if (httpURLConnection.getResponseCode() == Constants.RESPONSE_CODE_SUCCESS) {
+                InputStreamReader inputStreamReader =
+                        new InputStreamReader(httpURLConnection.getInputStream());
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: ERROR: [I18N][DM_DEFAULT_ENCODING] Found reliance on default encoding in com.framgia.movie.data.remote.DownloadJsonAsync.doInBackground(String[]): new java.io.InputStreamReader(InputStream)
http://findbugs.sourceforge.net/bugDescriptions.html#DM_DEFAULT_ENCODING

File: app/src/main/java/com/framgia/movie/data/model/Movie.java (hunk 1 above)
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/model/Movie.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.List'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+                BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
+                StringBuilder stringBuilder = new StringBuilder();
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                    stringBuilder.append(line).append("\n");
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: WARNING: [AppendCharacterWithChar] Avoid appending characters as strings in StringBuffer.append.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AppendCharacterWithChar

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+                }
+                httpURLConnection.disconnect();
+                bufferedReader.close();
+                return stringBuilder.toString();
+            }
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/com/framgia/movie/data/remote/MovieApi.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    private static final String KEY_API = "8ec3fbf1c1b06d940e29c592421917ae";
+    private static final String URL = "https://api.themoviedb.org/3";
+    public static final String URL_IMAGE = "https://image.tmdb.org/t/p/w500";
+
+    public static String getApiPopular(int page) {
+        return URL + "/movie/popular?api_key=" + KEY_API + "&language=en-US&page=" + page;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/MovieApi.java]: WARNING: [AvoidDuplicateLiterals] The String literal "&language=en-US&page=" appears 4 times in this file; the first occurrence is on line 14
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/main/java/com/framgia/movie/data/remote/ParceJson.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+                String posterPath = jsonObjectResult.getString(KEY_JSON_POSTER_PATH);
+                String packdropPath = jsonObjectResult.getString(KEY_JSON_BACKDROP_PATH);
+                String releaseDate = jsonObjectResult.getString(KEY_JSON_RELEASE_DATE);
+                Movie movie = new Movie(id, title, overview, voteAverage, posterPath, packdropPath,
+                        releaseDate);
+                movieList.add(movie);
+            }
+            iFace.ParcelJsonSuccess(movieList);
+        } catch (JSONException e) {
+            e.printStackTrace();
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/ParceJson.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+ * Created by TiepNguyen on 25-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class MainAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private static final int VIEW_TYPE_LOADING = 0;
+    private static final int VIEW_TYPE_ITEM = 1;
+    private List<Movie> mMovieList;
+    private Context mContext;
+    private boolean isLoading;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isLoading has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+    private LinearLayoutManager mLLManager;
+    private MainIFace mMainIFace;
+    private int page = 1;
+    private RecyclerView.OnScrollListener mOnScrollListener = new RecyclerView.OnScrollListener() {
+        @Override
+        public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+            int countItem = mLLManager.getItemCount();
+            int positionItemLastScroll = mLLManager.findLastCompletelyVisibleItemPosition();
+            if (countItem <= positionItemLastScroll + 5) {
+                if (!isLoading()) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: WARNING: [CollapsibleIfStatements] These nested if statements could be combined
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#CollapsibleIfStatements

File: app/src/main/java/com/framgia/movie/data/model/Movie.java (hunk 1 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/model/Movie.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/MovieApi.java (hunk 2 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/MovieApi.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/ParceJson.java (hunk 3 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/ParceJson.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 4 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 5 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java (hunk 6 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java (hunk 7 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java]: fix

File: app/src/main/java/com/framgia/movie/data/model/Movie.java (hunk 1 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/model/Movie.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java (hunk 9 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java (hunk 10 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java (hunk 11 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/MovieApi.java (hunk 12 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/MovieApi.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/ParceJson.java (hunk 13 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/ParceJson.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 14 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 15 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/MovieApi.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
+package com.framgia.movie.data.remote;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class MovieApi {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/MovieApi.java]: WARNING: Class MovieApi should be declared as final.

File: app/src/main/java/com/framgia/movie/data/remote/ParseJson.java [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+import org.json.JSONObject;
+
+import static android.content.ContentValues.TAG;
+
+/**
+ * Created by TiepNguyen on 25-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class ParseJson {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/ParseJson.java]: WARNING: Class ParseJson should be declared as final.

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+            MainIFace mainIFace) {
+        mContext = context;
+        mMovieList = movieList;
+        isLoading = false;
+        mLLManager = (LinearLayoutManager) recyclerView.getLayoutManager();
+        recyclerView.addOnScrollListener(mOnScrollListener);
+        mMainIFace = mainIFace;
+    }
+
+    public boolean getLoading() {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: WARNING: [BooleanGetMethodName] A getX() method which returns a boolean should be named isX()
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#BooleanGetMethodName

File: app/src/main/java/com/framgia/movie/data/remote/MovieApi.java (hunk 16 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/MovieApi.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/ParseJson.java (hunk 17 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/ParseJson.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 18 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+
+public class MainAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private static final int VIEW_TYPE_LOADING = 0;
+    private static final int VIEW_TYPE_ITEM = 1;
+    private List<Movie> mMovieList;
+    private Context mContext;
+    private boolean mIsLoading;
+    private LinearLayoutManager mLLManager;
+    private MainIFace mMainIFace;
+    private int page = 1;
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: page -> mPage

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+ * Created by TiepNguyen on 25-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class MainAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private static final int VIEW_TYPE_LOADING = 0;
+    private static final int VIEW_TYPE_ITEM = 1;
+    private List<Movie> mMovieList;
+    private Context mContext;
+    private boolean mIsLoading;
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: mIsLoading -> isLoading (đối với boolean thì dùng is)

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 19 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 21]
```diff
@@ -1,28 +1,90 @@
...
+    private MainAdapter mMainAdapter;
 
     @Override
-    public void onCreate(@Nullable Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_main, container, false);
+        mRecyclerMain = view.findViewById(R.id.recycler_main);
+        executeApiForTab(1);
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: hard code

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 22]
```diff
@@ -1,28 +1,90 @@
...
+
+    @Override
+    public void parseJsonSuccess(List<Movie> movieList) {
+        if (mMovieList == null) {
+            mMovieList = movieList;
+            GridLayoutManager gridLayoutManager = new GridLayoutManager(getContext(), 2);
+            gridLayoutManager.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
+                @Override
+                public int getSpanSize(int position) {
+                    if (mMainAdapter.isLoading() && position > mMovieList.size() - 2) {
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: same

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 23]
```diff
@@ -1,28 +1,90 @@
...
+                        return 2;
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: same

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 24]
```diff
@@ -1,28 +1,90 @@
...
+                    }
+                    return 1;
+                }
+            });
+            mRecyclerMain.setLayoutManager(gridLayoutManager);
+            mMainAdapter = new MainAdapter(getContext(), mMovieList, mRecyclerMain, this);
+            mRecyclerMain.setAdapter(mMainAdapter);
+        } else {
+            mMovieList.remove(mMovieList.size() - 1);
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: same

File: app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java [hunk 25]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.framgia.movie.data.model.Movie;
+import java.util.List;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public interface MainIFace {
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainIFace.java]: sửa tên MainIFace đi, đọc tối nghĩa quá

File: app/src/main/res/layout/item_movie.xml [hunk 26]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    >
+    <ImageView
+        android:id="@+id/image_movie"
```
Comment (ThuanPx) [inline app/src/main/res/layout/item_movie.xml]: đổi lại id đừng dùng _ nữa đi e

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java [hunk 27]
```diff
@@ -0,0 +1,96 @@
...
+                        return spanDefault;
+                    }
+                    return spanLoading;
+                }
+            });
+            mRecyclerMain.setLayoutManager(gridLayoutManager);
+            mMainAdapter = new MainAdapter(getContext(), mMovieList, mRecyclerMain, this);
+            mRecyclerMain.setAdapter(mMainAdapter);
+        } else {
+            int positionItemProgress = mMovieList.size() -1;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java]: ERROR: '-' is not followed by whitespace.

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java [hunk 28]
```diff
@@ -0,0 +1,96 @@
...
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+
+public class ExecuteJsonFragment extends Fragment implements ExecuteJsonIFace {
+    private RecyclerView mRecyclerMain;
+    private List<Movie> mMovieList;
+    private MainAdapter mMainAdapter;
+    private final int spanDefault = 2;
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java]: đặt tên sai

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+    private final int spanLoading = 1;
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java]: same

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonIFace.java [hunk 30]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.framgia.movie.data.model.Movie;
+import java.util.List;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public interface ExecuteJsonIFace {
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonIFace.java]: interface thì đặt tên theo kiểu XXXListener

File: app/src/main/res/layout/item_movie.xml [hunk 31]
```diff
@@ -0,0 +1,27 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    >
+    <ImageView
+        android:id="@+id/imageMovie"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:adjustViewBounds="true"
+        android:contentDescription="@string/text_content_description"
+        android:src="@drawable/image_demo"
```
Comment (ThuanPx) [inline app/src/main/res/layout/item_movie.xml]: sai ten icon nè

Comment (NguyenChanhTiep) [reply app/src/main/res/layout/item_movie.xml]: fix

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java [hunk 32]
```diff
@@ -0,0 +1,61 @@
...
+import java.nio.charset.Charset;
+
+import static android.content.ContentValues.TAG;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class DownloadJsonAsync extends AsyncTask<String, Void, String> {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: ``DownloadJsonAsync`` -> ``FetchDataFromServerTask``

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java [hunk 33]
```diff
@@ -0,0 +1,61 @@
...
+    private ExecuteJsonListener mExecuteJsonListener;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: ``ExecuteJsonListener`` -> ``FetchDataFromServerCallback``

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java [hunk 34]
```diff
@@ -0,0 +1,96 @@
...
+import com.framgia.movie.ui.adapter.MainAdapter;
+import com.framgia.movie.ui.adapter.ViewPagerAdapter;
+import com.framgia.movie.util.Constants;
+import java.util.List;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+
+public class ExecuteJsonFragment extends Fragment implements ExecuteJsonListener {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java]: ``ExecuteJsonFragment`` -> ``ListMovieFragment``

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java [hunk 35]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.framgia.movie.data.model.Movie;
+import java.util.List;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public interface ExecuteJsonListener {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java]: ``ExecuteJsonListener`` -> ``FetchDataFromServerCallback``

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java [hunk 36]
```diff
@@ -0,0 +1,15 @@
...
+    void downloadSuccess(String data);
+    void parseJsonSuccess(List<Movie> movieList);
+    void requestLoadMore(int page);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java]: remove ``void requestLoadMore(int page);`` from this interface
replace by declare in new interface with name ``LoadMoreListener``

File: app/src/main/java/com/framgia/movie/util/Constants.java [hunk 37]
```diff
@@ -12,4 +12,5 @@ private Constants() {
 
     public static final String KEY_SEARCH = "KEY_SEARCH";
     public static final String KEY_TAB_FRAGMENT_ARGUMENTS = "KEY_TAB_FRAGMENT_ARGUMENTS";
+    public static final int RESPONSE_CODE_SUCCESS = 200;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/util/Constants.java]: remove this line!
should use ``HttpURLConnection.HTTP_OK``

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java (hunk 32 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java (hunk 33 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/data/remote/DownloadJsonAsync.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java (hunk 34 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonFragment.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java (hunk 35 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java (hunk 36 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/ExecuteJsonListener.java]: fix

File: app/src/main/java/com/framgia/movie/util/Constants.java (hunk 37 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/util/Constants.java]: fix

File: app/src/main/java/com/framgia/movie/data/remote/FetchDataFromServerTask.java [hunk 38]
```diff
@@ -0,0 +1,61 @@
+package com.framgia.movie.data.remote;
+
+import android.os.AsyncTask;
+import android.util.Log;
+import com.framgia.movie.ui.fragment.FetchDataFromServerCallback;
+import com.framgia.movie.util.Constants;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/FetchDataFromServerTask.java]: ERROR: Unused import: com.framgia.movie.util.Constants. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java [hunk 39]
```diff
@@ -0,0 +1,133 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import com.framgia.movie.R;
+import com.framgia.movie.data.model.Movie;
+import com.framgia.movie.data.remote.MovieApi;
+import com.framgia.movie.ui.fragment.FetchDataFromServerCallback;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: ERROR: Unused import: com.framgia.movie.ui.fragment.FetchDataFromServerCallback. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/movie/ui/fragment/ListMovieFragment.java [hunk 40]
```diff
@@ -0,0 +1,96 @@
...
+import com.framgia.movie.ui.adapter.MainAdapter;
+import com.framgia.movie.ui.adapter.ViewPagerAdapter;
+import com.framgia.movie.util.Constants;
+import java.util.List;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+
+public class ListMovieFragment extends Fragment implements FetchDataFromServerCallback, LoadMoreListener {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/ListMovieFragment.java]: ERROR: Line is longer than 100 characters (found 106).

File: app/src/main/java/com/framgia/movie/data/remote/FetchDataFromServerTask.java (hunk 38 above)
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/data/remote/FetchDataFromServerTask.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.util.Constants'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java (hunk 39 above)
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/MainAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.ui.fragment.FetchDataFromServerCallback'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Pull request merged**